### PR TITLE
Remove option state that is redundant with existing field

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

File: src/Workspaces/Core/Portable/Options/GlobalOptionService.cs [hunk 1]
```diff
@@ -270,7 +270,7 @@ public SerializableOptionSet GetSerializableOptionsSnapshot(ImmutableHashSet<str
             var changedOptionsKeysSerializable = _changedOptionKeys
                 .Where(key => serializableOptions.Contains(key.Option) && (!key.Option.IsPerLanguage || languages.Contains(key.Language!)))
                 .ToImmutableHashSet();
-            return new SerializableOptionSet(languages, optionService, serializableOptions, serializableOptionValues, changedOptionsKeysSerializable);
+            return new SerializableOptionSet(languages, optionService, serializableOptionValues, changedOptionsKeysSerializable);
```
Comment (CyrusNajmabadi) [inline src/Workspaces/Core/Portable/Options/GlobalOptionService.cs:273]: this is the only location i can find where we really genuinely flow in this information.  And it is genuinely redundant with serializableOptionValues (the latter is computed from the former).

**Review by tmat — APPROVED** [review]
:shipit:

**Comment by CyrusNajmabadi** [comment]
@mavasani it might be easier to review this PR in isolation.

File: src/Workspaces/Core/Portable/Options/GlobalOptionService.cs (hunk 1 above)
Comment (CyrusNajmabadi) [reply src/Workspaces/Core/Portable/Options/GlobalOptionService.cs:273]: @mavasani see comment here: https://github.com/dotnet/roslyn/pull/56655#discussion_r715770778

**Review by mavasani — APPROVED** [review]
LGTM after looking through all the surrounding code in GlobalOptionService, thanks!

**Pull request merged**